### DISCUSSION TITLE
How DynamicNode capitalize the first letter, when input?

### DISCUSSION THREAD

**Discussion by YanTree** [L3MON4D3/LuaSnip#1087] [Q&A] [answered]
Slot `2` `3`, the DynamicNode should return a string when input, but capitalize the first letter.

I tried blow code, but it's not work for me, can anyone could give me some advice?
``` lua
s(
  "sample",
  fmt("half4 {} = SAMPLE_TEXTURE2D(_{}, sampler_{}, input.uv.xy);", {
    i(1, "baseMap"),
    d(2, function(args)
      return sn(nil, {
        i(1, l(l._1:sub(1, 1):upper()..l._1:sub(2, -1), args[1]))
      })
      end,
    {1}),
    d(3, function(args)
      return sn(nil, {
        i(1, l(l._1:sub(1, 1):upper()..l._1:sub(2, -1), args[1]))
      })
      end,
    {1}),
})),

```

**Comment by L3MON4D3** [comment, marked as answer]
Mhhh you don't need the `l(...)` inside the `i(...)`, try `return i(1, args[1][1]:sub(1,1):upper .. args[1][1]:sub(2,-1))` (or, replace the whole `d(2, ...)` with `dl(2, l._1:sub(1, 1):upper()..l._1:sub(2, -1), {1})`, something like that should work too)

**Reply by YanTree** [reply]
wow! thx for reply, this problem has troubled me for a whole day.

1. `dl(2, l._1:sub(1, 1):upper()..l._1:sub(2, -1), {1})` it works ✅
2. `return i(1, args[1][1]:sub(1,1):upper .. args[1][1]:sub(2,-1))` will show one diagnostics with 'Missed symbol '('.' ⭕

So finally code as below:
``` shaderlab
s(
  "sample",
  fmt("half4 {} = SAMPLE_TEXTURE2D(_{}, sampler_{}, input.uv.xy);", {
    i(1, "baseMap"),
    dl(2, l._1:sub(1, 1):upper()..l._1:sub(2, -1), {1}),
    dl(3, l._1:sub(1, 1):upper()..l._1:sub(2, -1), {1}),
})),
```

**Reply by L3MON4D3** [reply]
Ah, 2. is just missing the parantheses after the `upper`.
But 1. is fancier anyway :D
Glad I could help :)
